examples/templates/azure-linux/main.tf

```diff
@@ -0,0 +1,227 @@
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "coder_workspace" "me" {
+}
+
+resource "coder_agent" "main" {
+  arch = "amd64"
+  os   = "linux"
+}
+
+locals {
+  prefix = "coder-${data.coder_workspace.me.owner}-${data.coder_workspace.me.name}"
+
+  userdata = templatefile("cloud-config.yaml.tftpl", {
+    username          = lower(substr(data.coder_workspace.me.owner, 0, 32))
+    init_script       = base64encode(coder_agent.main.init_script)
+    coder_agent_token = coder_agent.main.token
+    hostname          = lower(data.coder_workspace.me.name)
+  })
+}
+
+resource "azurerm_resource_group" "main" {
+  name     = "${local.prefix}-resources"
+  location = var.location
+
+  tags = {
+    Name              = "coder-${data.coder_workspace.me.owner}-${data.coder_workspace.me.name}"
+    Coder_Provisioned = "true"
+  }
+}
+
+// Uncomment here and in the azurerm_network_interface resource to obtain a public IP
+#resource "azurerm_public_ip" "main" {
+#  name                = "publicip"
+#  resource_group_name = azurerm_resource_group.main.name
+#  location            = azurerm_resource_group.main.location
+#  allocation_method   = "Static"
+#
+#  tags = {
+#    Name = "coder-${data.coder_workspace.me.owner}-${data.coder_workspace.me.name}"
+#    Coder_Provisioned = "true"
+#  }
+#}
+
+resource "azurerm_virtual_network" "main" {
+  name                = "network"
+  address_space       = ["10.0.0.0/24"]
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+
+  tags = {
+    Name              = "coder-${data.coder_workspace.me.owner}-${data.coder_workspace.me.name}"
+    Coder_Provisioned = "true"
+  }
+}
+
+resource "azurerm_subnet" "internal" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.main.name
+  virtual_network_name = azurerm_virtual_network.main.name
+  address_prefixes     = ["10.0.0.0/29"]
+}
+
+resource "azurerm_network_interface" "main" {
+  name                = "nic"
+  resource_group_name = azurerm_resource_group.main.name
+  location            = azurerm_resource_group.main.location
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.internal.id
+    private_ip_address_allocation = "Dynamic"
+    // Uncomment for public IP address as well as azurerm_public_ip resource above
+    //public_ip_address_id = azurerm_public_ip.main.id
+  }
+
+  tags = {
+    Name              = "coder-${data.coder_workspace.me.owner}-${data.coder_workspace.me.name}"
+    Coder_Provisioned = "true"
+  }
+}
+
+resource "azurerm_managed_disk" "home" {
+  create_option        = "Empty"
+  location             = azurerm_resource_group.main.location
+  name                 = "home"
+  resource_group_name  = azurerm_resource_group.main.name
+  storage_account_type = "StandardSSD_LRS"
+  disk_size_gb         = var.home_size
+}
```

Comment:
We should at minimum add the following tags:
- Coder_Provisioned = true

Adding a "name" tag seems to be redundant here though.